README.md

```diff
@@ -31,4 +31,49 @@ Example on a local Jupyter installation.
 
 On JupyterHub,
 
-  https://jupyterhub-dev.uwrl.usu.edu/hub/user-redirect/hs-pull?id=8caa62c46c424a818899ebeca6f30a83&start=spiro3D.ipynb
\ No newline at end of file
+  https://jupyterhub-dev.uwrl.usu.edu/hub/user-redirect/hs-pull?id=8caa62c46c424a818899ebeca6f30a83&start=spiro3D.ipynb
+
+## Development
+
+Setup development environment
+
+```
+# clone git repo
+git clone git@github.com:hydroshare/nbfetch.git && cd nbfetch
+
+# create and activate python environment
+python3 -m venv venv
+source venv/bin/activate
+pip install --upgrade pip
+
+# install nbfetch in edit mode
+pip install -e .
+
+# enable jupyter_server extension
+jupyter server extension enable --py nbfetch
+
+# start jupyter lab
+jupyter lab
+```
+
+`nbfetch` requires HydroShare user authentication (username, password) or authorization (OAuth2) to
+function. HydroShare user authentication can be provided in two ways, by file or via environment
+variables (see below). At this time, we do not have guidance for supplying using authorization
+(OAuth2) in a development environment.
```

Comment:
In DSP, we have a test client application registered and save the credentials with the default environment configuration.

https://github.com/cznethub/dspback/blob/develop/.env#L7-L10

We could setup something similar here or provide instructions on how to setup a new client application and copy the client key/secret into a local configuration.

Reply:
That sounds perfect to me. Can you move this to an issue and maybe we can work through the details there? I have worked out configuring unit tests, just havent gotten around to writing any meaningful additions. What you mentioned above plus unit tests, so like a good future milestone.